docs/src/architecture/data-model.md

```diff
@@ -0,0 +1,152 @@
+
+### Measurements
+  * **measurements** carry values from physical **Sensors**[^1] or a device's **Domain Application**[^1];
+    e.g. voltage and current of an electricity meter, or current state of the manufacturing control process
+  * **elements of a measurement:**
+    - `type name`, a string that identifies the measurement uniquely in context of a device
+    - `samples`, carry a single value or set of values, all taken at a single point in time
+      - `values` can be a mix of numbers, strings or booleans
+        - optionally each value can have a `type name`
+      - `timestamp` (conform to ISO 8601), that indicates when values were sampled
+    - TODO: not yet supported: list of units per sample's values
+    - TODO: not yet supported: source
+  * **behaviour of a measurement:**
+    - thin-edge does not store any historical sampled values for measurements
+    - there is no initialization-value for measurements; i.E. a measurement is not visible on thin-edge before the 1st sample was sent to thin-edge
+
+### Events
+  * **events** are notifications that something happened on a device's environment or software system;
+    e.g. a sensor[^1] detected something like a door has been closed,
+    or a system notification that e.g. a user has started an ssh session
+  * **elements of an event:**
+    - `type name`, a string that identifies the event uniquely in context of a device
+    - `event text`, that carries a human readable event-message
+      - must be UTF-8 encoded
+    - `timestamp` (conform to ISO 8601), that indicates when the event has occured
+    - an event can optionally contain any custom-specific extra-information
+    - TODO: not yet supported: attachements
+  * **behaviour of an event:**
+    - thin-edge does not store any historical occurrences for events
+
+### Alarms
+  * **alarms** are notifications about some critical behaviour of the device's environment or software system;
+    e.g. when a temperature sensor detects a temperature went out of its valid range
+  * **elements of an alarm:**
+    - `type name`, a string that identifies the alarm uniquely in context of a device
+    - alarm text, that carries an alarm message
+      - must be UTF-8 encoded
+    - severity, that is one of  `critical`, `major`, `minor` or `warning`
+    - `timestamp` (conform to ISO 8601), that indicates when the alarm has occured
+    - an alarm can optionally contain any custom-specific extra-information
+  * **behaviour of an alarm:**
+    - thin-edge does not store any historical occurrences for alarms
+    - **alarms** are stateful; i.E. once raised, an **alarm** is active until it was explicitely cleared by the device's software or the cloud
+
+[^1]: details see "Domain Model" appendix [Device Domain](./domain-model.md#device-overview) -->
+
+## Use of MQTT
+
+**thin-edge** expects the MQTT broker [mosquitto](https://mosquitto.org/) being installed on the device.
```

Comment:
It will be better to keep it a bit more generic as we will be supported external mqtt brokers in the future, so the broker will not be installed on the device.

```suggestion
**thin-edge** expects the MQTT broker [mosquitto](https://mosquitto.org/) to be available on the device.
```